### DISCUSSION TITLE
is there a way to deserialise a JSON string of Bucket to the Bucket object in java

### DISCUSSION THREAD

**Discussion by nevingeorgesunny** [vladimir-bukhtoyarov/bucket4j#255] [Q&A] [answered]
Here is the Json string
`{"timeMeter":{"wallClockBased":true},"configuration":{"bandwidths":[{"capacity":2,"initialTokens":2,"refillPeriodNanos":100000000000,"refillTokens":2,"refillIntervally":true,"timeOfFirstRefillMillis":-9223372036854776000,"useAdaptiveInitialTokens":false,"id":null,"gready":false,"intervallyAligned":false}]},"availableTokens":1}`

A little bit of context of what I am trying to do

- I have created a bucket object for rate-limiting and currently saving in redis
- But not able to retrieve this from Redis as I need to create a custom deserialiser. 
- Im using Jackson lib for deserialising in my project which needs annotations.
- is there any custom deserialiser in the project to take the input json and return the bucket object ?

**Comment by vladimir-bukhtoyarov** [comment]
No, it is impossible. Several constructors were explicitly declared with "private" visibility in order to hide it from end users, so instantiations of mostly Bucket4j objects from outside is not supported.

However, soon will be added ability that was discussed there [#250 How to keep state between app restarts?](https://github.com/vladimir-bukhtoyarov/bucket4j/discussions/250#discussioncomment-2567006).

**Comment by vladimir-bukhtoyarov** [comment, marked as answer]
@nevingeorgesunny 
In the scope of release 7.5.0 were added following features:
* [Sereliaze](https://github.com/vladimir-bukhtoyarov/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/local/LocalBucket.java#L61) / [deserialize](https://github.com/vladimir-bukhtoyarov/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/local/LocalBucket.java#L72) to binary array.
* [Sereliaze](https://github.com/vladimir-bukhtoyarov/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/local/LocalBucket.java#L81) / [deserialize](https://github.com/vladimir-bukhtoyarov/bucket4j/blob/master/bucket4j-core/src/main/java/io/github/bucket4j/local/LocalBucket.java#L92) to hashmap that holds only basic java types in result this map can be transparently serialized/ deserialized by any JSON serialization library.
